### PR TITLE
Add candidates and applications by status exports

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

This is the first of many PRs to add exports for each of the MonthlyStatistics tables. These will at 00:01 after the `UpdateMonthlyStatisticsReport` worker has run.

I've tried to build it in an agnostic way which relies on the keys to generate munge the totals row into the rows hash. This should mean that the helper can be used for all the tables going forward.

## Changes proposed in this pull request

- Add an export for candidates by status
- Add an export for applications by status
- Add a reusable helper which extracts out common logic 

## Guidance to review

Out of scope 
- Running these on the first of each month via clock
- Adding them to the monthly report UI (need to discuss what this looks like with Frankie)

## Link to Trello card

https://trello.com/c/OqvHtjG2/4111-add-exports-for-each-of-the-monthly-statistics-tables-catch-all

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/helpers/external_report_export_helper.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module ExternalReportExportHelper
```
Comment (TobyRet) [inline app/helpers/external_report_export_helper.rb:1]: 'MonthlyStatisticsExportHelper`?

File: app/services/support_interface/external_report_exports/applications_by_status_export.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module SupportInterface
```
Comment (TobyRet) [inline app/services/support_interface/external_report_exports/applications_by_status_export.rb]: The monthly stats code does not currently live in the 'support' namespace as it is a publicly accessible. At the time it did not feel like its home should be within the candidate or support namespaces ... 
Oh ... but I guess the exports are only made available via the support UI?

Comment (davidgisbey) [reply app/services/support_interface/external_report_exports/applications_by_status_export.rb]: Yea the exports will live in the Support UI. It's a good question about where the dashboard should sit actually 🤔

File: app/helpers/external_report_export_helper.rb (hunk 1 above)
Comment (davidgisbey) [reply app/helpers/external_report_export_helper.rb]: Yea have done. I'll. update the naming in the other exports as well as part of this PR

File: app/helpers/monthly_statistics_export_helper.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+private
+
+  def merge_totals(hash, totals)
+    hash.map do |key, _value|
+      [key, totals[hash.find_index { |k, _v| k == key }]]
+    end
+  end
+end
```
Comment (stevehook) [inline app/helpers/monthly_statistics_export_helper.rb:16]: I found this a bit hard to follow. I had a go at simplifying it and I'm not sure I've made it any clearer, just shorter:

```
module MonthlyStatisticsExportHelper
  def merge_rows_and_totals(data)
    column_names = data[:rows].first.drop(1).map(&:first)
    totals = data[:column_totals]

    data[:rows] + [{ data[:rows].first.keys.first => 'Total' }.merge!(column_names.zip(totals).to_h)]
  end
end
```

**Review by stevehook — COMMENTED** [review]
This looks really good to me. Made one code suggestion inline but not convinced it improves anything so totally optional.

I get an error when I try to run either of the new reports from the data directory - e.g. `'monthly_statistics_candidates_by_status' is not a valid export_type`. If that's being handled in a downstream PR then it's all good to merge.

File: app/helpers/monthly_statistics_export_helper.rb (hunk 3 above)
Comment (davidgisbey) [reply app/helpers/monthly_statistics_export_helper.rb:16]: Yea i think this is def better. Thanks Steve!

TIL about zip

**Comment by davidgisbey** [comment]
@stevehook good spot, that is fixed in a subsequent PR

**Review by stevehook — APPROVED** [review]
